### PR TITLE
198 Include user's card labels when generating the report

### PR DESCRIPTION
**Pull request by kartikeyakirar**
this PR fixes https://github.com/insightsengineering/teal.goshawk/issues/240

this is follow-up after https://github.com/insightsengineering/teal.reporter/pull/219

### CODE REVIEW

**Comment by kartikeyakirar** [comment]
CMD checks are failing because of  `mcr` package is archived from CRAN

**Comment by m7pr** [comment]
@kartikeyakirar please take out card_template and move it to teal.reporter.
https://github.com/insightsengineering/teal.modules.general/pull/584#issuecomment-1750699411

**Comment by m7pr** [comment]
@kartikeyakirar please take out card_template and move it to teal.reporter.
https://github.com/insightsengineering/teal.modules.general/pull/584#issuecomment-1750699411

File: R/tm_g_gh_boxplot.R [hunk 1]
```diff
@@ -548,20 +548,24 @@ srv_g_boxplot <- function(id,
...
-            "\nSelect an X-axis Variable:",
-            input$xaxis_var
+      card_fun <- function(comment, label) {
+        constraint_description <- paste(
+          "\nFacet By:",
+          if (length(input$facet_var) != 0) input$facet_var else "None",
+          "\nSelect an X-axis Variable:",
+          input$xaxis_var
+        )
+        card <- report_card_template(
```
Comment (m7pr) [inline R/tm_g_gh_boxplot.R:558]:
```suggestion
        card <- teal::report_card_template(
```

**Review by m7pr — CHANGES_REQUESTED** [review]
please append teal:: before report_card_template so it's easier for debugging in the future

File: R/tm_g_gh_boxplot.R (hunk 1 above)
Comment (m7pr) [reply R/tm_g_gh_boxplot.R:558]: cc @kartikeyakirar and there is more than one place for this

File: R/utils.R [hunk 2]
```diff
@@ -31,3 +31,49 @@ plots_per_row_validate_rules <- function(required = TRUE) {
...
+#'   - constraint_range_min (`numeric(1)`) the minimum constraint range value.
+#'   - constraint_range_max (`numeric(1)`) the maximum constraint range value.
+#' @param constraint_description (`character(1)`)  description of the constraints.
+#' @param style (`character(1)`)  style of the constraint text block.
+#'   options: `default`, `verbatim` (default is `default`).
+#'
+#' @return (`TealReportCard`) populated with a title, description, and filter state
+#'
+#' @keywords internal
+report_card_template <- function(title,
```
Comment (m7pr) [inline R/utils.R:48]: @kartikeyakirar this is not a good idea to have the same name of the function in 2 packages. Please rename to `report_card_template_goshawk`

**Review by m7pr — CHANGES_REQUESTED** [review]

File: R/tm_g_gh_boxplot.R (hunk 1 above)
Comment (m7pr) [reply R/tm_g_gh_boxplot.R:558]: ok it looks like it is a wrapper provided in teal.goshawk. You see how this name was confusing for me as it is the same as in teal. So please follow my advise from https://github.com/insightsengineering/teal.goshawk/pull/241#discussion_r1356596742

**Comment by kartikeyakirar** [comment]
@m7pr  there is no namesapce call because  its internal function I have defined `report_card_template` here in the `utils`, and it makes use of `teal::report_card_template`. https://github.com/insightsengineering/teal.goshawk/blob/198_card_labels%40main/R/utils.R#L35-L79

**Comment by m7pr** [comment]
> CMD checks are failing because of mcr package is archived from CRAN
Is back on CRAN

File: R/tm_g_gh_boxplot.R (hunk 1 above)
Comment (kartikeyakirar) [reply R/tm_g_gh_boxplot.R:558]: https://github.com/insightsengineering/teal.goshawk/pull/241#issuecomment-1759322644

**Comment by m7pr** [comment]
https://github.com/insightsengineering/teal.goshawk/pull/241#discussion_r1356597506

**Review by m7pr — APPROVED** [review]

File: R/utils.R [hunk 3]
```diff
@@ -31,3 +31,49 @@ plots_per_row_validate_rules <- function(required = TRUE) {
     shinyvalidate::sv_gt(0, message_fmt = msg)
   )
 }
+
+#' Template function to generate reporter card for `teal.goshawk`
```
Comment (m7pr) [inline R/utils.R:35]: @kartikeyakirar this is a title of RD file. Can you make it TITLE case . Can you use the same structure of teal? https://github.com/insightsengineering/teal/pull/933#discussion_r1356494205 Why am I making the same advise twice?

**Review by m7pr — CHANGES_REQUESTED** [review]

File: R/utils.R (hunk 3 above)
Comment (kartikeyakirar) [reply R/utils.R]: There was oversight. sorry about that.

**Review by m7pr — APPROVED** [review]

**Pull request merged**